platform-tests/src/test/java/org/junit/platform/engine/support/hierarchical/ResourceLockTests.java

```diff
@@ -12,68 +12,78 @@
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.platform.engine.support.hierarchical.ExclusiveResource.GLOBAL_READ;
+import static org.junit.platform.engine.support.hierarchical.ExclusiveResource.GLOBAL_READ_WRITE;
+import static org.junit.platform.engine.support.hierarchical.ExclusiveResource.singleton;
 
 import java.util.List;
+import java.util.NavigableSet;
+import java.util.TreeSet;
 import java.util.concurrent.locks.ReentrantLock;
 
 import org.junit.jupiter.api.Test;
+import org.junit.platform.engine.support.hierarchical.ExclusiveResource.LockMode;
 
-@SuppressWarnings("resource")
 class ResourceLockTests {
 
 	@Test
 	void nopLocksAreCompatibleWithEverything() {
 		var nopLock = NopLock.INSTANCE;
 
+		assertTrue(nopLock.isCompatible(nopLock));
 		assertTrue(nopLock.isCompatible(NopLock.INSTANCE));
-		assertTrue(nopLock.isCompatible(new SingleLock(anyReentrantLock())));
-		assertTrue(nopLock.isCompatible(new SingleLock.GlobalReadLock(anyReentrantLock())));
-		assertTrue(nopLock.isCompatible(new SingleLock.GlobalReadWriteLock(anyReentrantLock())));
-		assertTrue(nopLock.isCompatible(new CompositeLock(List.of(anyReentrantLock()))));
+		assertTrue(nopLock.isCompatible(new SingleLock(anyResource(), anyReentrantLock())));
+		assertTrue(nopLock.isCompatible(new CompositeLock(singleton(anyResource()), List.of(anyReentrantLock()))));
 	}
 
 	@Test
-	void singleLocksAreIncompatibleWithNonNopLocks() {
-		var singleLock = new SingleLock(anyReentrantLock());
+	void singleLocksAreIncompatibleWithOtherLocksThatCanPotentiallyCauseDeadlocks() {
+		ExclusiveResource resourceB = resource("B", LockMode.READ);
+		var singleLock = new SingleLock(resourceB, anyReentrantLock());
```

Comment:
🧐 The name suggests that we check for incompatibility, yet there are compatible assertions here. I would split it accordingly, even making it data-driven for easier comprehension.

Applies to the other tests as well.

Reply:
I think it's easier to see compatible vs. incompatible this way because the example are directly next to each other.